### PR TITLE
Web3 integration

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

**Comment by marekkirejczyk** [comment]
1. fix tests
2. remove console.logs

File: universal-login-web3/lib/ULWeb3Provider.ts [hunk 1]
```diff
@@ -46,13 +48,15 @@ export class ULWeb3Provider implements Provider {
...
     ensDomains,
+    applicationInfo,
     uiInitializer = initUi,
     storageService = new StorageService(),
   }: ULWeb3ProviderOptions) {
     this.provider = provider;
     this.sdk = new UniversalLoginSDK(
       relayerUrl,
       new providers.Web3Provider(this.provider as any),
+      applicationInfo && {applicationInfo},
```
Comment (marekkirejczyk) [inline universal-login-web3/lib/ULWeb3Provider.ts]: extract 
```ts
sdkConfigFor(applicationInfo) => applicationInfo && {applicationInfo}
```

File: universal-login-web3/test/Web3Picker.tsx [hunk 2]
```diff
@@ -3,38 +3,61 @@ import sinon from 'sinon';
...
+    web3Picker.show = sinon.stub().returns({waitForPick: async () => {}});
+    web3Strategy.web3picker = web3Picker;
+  });
+
   it('Pick ul provider', () => {
-    web3Strategy.send(jsonRpcReq, () => {});
     web3Picker.setProvider(universalLoginProviderFactory.name);
-    expect(createSpy.calledOnce).to.be.true;
+    expect(createStub.calledOnce).to.be.true;
+    web3Strategy.send(jsonRpcReq, () => {});
```
Comment (JustynaBroniszewska) [inline universal-login-web3/test/Web3Picker.tsx]:
```ts
web3Strategy.send(jsonRpcReq, callback);
expect(callback).to.have.been.called;
```

File: universal-login-web3/lib/Web3Strategy.ts [hunk 3]
```diff
@@ -5,13 +5,24 @@ import {Web3ProviderFactory} from './models/Web3ProviderFactory';
...
 
   send(payload: JsonRPCRequest, callback: Callback<JsonRPCResponse>) {
-    return this.currentProvider.send(payload, callback);
+    switch (payload.method) {
+      case 'eth_sendTransaction':
+      case 'eth_sendRawTransaction':
+      case 'eth_accounts':
+      case 'eth_sign':
+      case 'personal_sign':
+        return this.currentProvider.send(payload, callback);
```
Comment (marekkirejczyk) [inline universal-login-web3/lib/Web3Strategy.ts]: `readProvider` -> `readProvider`
`currentProvider` ->`mutatingProvider`

**Pull request merged**